app/celer-g4/DetectorConstruction.cc

```diff
@@ -97,6 +105,44 @@ G4VPhysicalVolume* DetectorConstruction::Construct()
         sd.enabled = false;
     }
 
+    // Setup options for the magnetic field
+    auto field_type = GlobalSetup::Instance()->GetFieldType();
+    if (field_type == "rzmap")
+    {
+        auto map_filename = GlobalSetup::Instance()->GetFieldFile();
+        if (map_filename.empty())
+        {
+            G4Exception("DetectorConstruction::Construct()",
+                        "",
+                        FatalException,
+                        "No field file was specified with /celerg4/fieldFile");
+        }
+        CELER_LOG_LOCAL(info) << "Using RZMapField with " << map_filename;
+
+        // Create celeritas::RZMapFieldParams from input
+        RZMapFieldInput rz_map;
+        std::ifstream(map_filename) >> rz_map;
+        field_params_ = std::make_shared<RZMapFieldParams>(rz_map);
+        mag_field_ = std::make_shared<RZMapMagneticField>(field_params_);
+
+        celeritas::app::GlobalSetup::Instance()->GetAlongStepOptions()
+            = celeritas::RZMapFieldAlongStepFactory([=] { return rz_map; });
+    }
+    else
+    {
+        G4ThreeVector field{0, 0, 0};
+        if (field_type == "uniform")
+        {
+            field = GlobalSetup::Instance()->GetMagFieldZTesla();
+        }
+        CELER_LOG_LOCAL(info)
+            << "Using a uniform field (0, 0, " << field[2] << ") in Tesla";
+        mag_field_ = std::make_shared<G4UniformMagField>(field);
+
+        celeritas::app::GlobalSetup::Instance()->GetAlongStepOptions()
+            = UniformAlongStepFactory([&] { return field; });
```

Comment:
This needs to be a copy, since the along-step method is being built after his function ends.
```suggestion
            = UniformAlongStepFactory([=] { return field; });
```